### PR TITLE
Kernel number conversion functions support "exception" kwarg

### PR DESCRIPTION
**Pull request by pje**
This was added in Ruby 2.6: https://bugs.ruby-lang.org/issues/12732

Implements #1060

### CODE REVIEW

**Comment by pje** [comment]
This causes a failure:

```
$ bazel test //test:test_PosTests/testdata/rbi/kernel
https://github.com/sorbet/sorbet/tree/master/rbi/core/kernel.rbi#L464: Malformed `sig`. Overloaded functions cannot have keyword arguments:  `exception` https://srb.help/5003
     464 |        exception: T::Boolean
                  ^^^^^^^^^
```

If we want to support this, seems like we'd have to get rid of overloading for these functions and always return a `nilable` result, which isn't ideal.

Do any of the sorbet folks have thoughts on this?

**Comment by DarkDimius** [comment]
> Overloaded functions cannot have keyword arguments:  `exception` https://srb.help/5003

it's not a fundamental restriction, but rather something that's hard to implement.

> T.nilable(Float)

I also feel that correctly marking those as sometimes returning `nil`, while being formally correct, will be super irritating. Do you know of this catching real bugs?

**Comment by pje** [comment]
> it's not a fundamental restriction, but rather something that's hard to implement.

OK, thanks for that clarification! 

> I also feel that correctly marking those as sometimes returning nil, while being formally correct, will be super irritating.

Yeah, I agree. :\ 

My only goal here was to get Sorbet to accept a valid line of Ruby. Currently if people have code like `Integer('123', exception: false)`, their only options are to rewrite it or to `typed: ignore` the entire file.

In lieu of support for keyword args in overloaded sigs, it sounds like we have two choices: 1) keep the current behavior, or 2) get rid of the overloading in these sigs to support the `exception` keyword.

Personally I'd vote for the latter—we'd be sacrificing some specificity in these type signatures, but that seems like a reasonable tradeoff to support a valid use of a commonly-used, standard language function. **I just pushed that change in the last commit to this branch**. WDYT, @DarkDimius?

File: rbi/core/kernel.rbi [hunk 1]
```diff
@@ -451,33 +451,33 @@ module Kernel
     params(
         initial: T.any(Integer, Float, Rational, BigDecimal, String),
         digits: Integer,
+        exception: T::Boolean
     )
     .returns(BigDecimal)
   end
-  def BigDecimal(initial, digits=0); end
+  def BigDecimal(initial, digits=0, exception: true); end
+  # As of 2.6, all these numeric conversion methods *can* return `nil` iff given the kwarg `exception: false`, which
```
Comment (DarkDimius) [inline rbi/core/kernel.rbi]: could you add a link to this PR as it has valuable discussion?

File: rbi/core/kernel.rbi [hunk 2]
```diff
@@ -451,33 +451,33 @@ module Kernel
...
     .returns(Complex)
   end
-  sig do
-    params(
-        x: String,
-    )
-    .returns(Complex)
-  end
-  def Complex(x, y=T.unsafe(nil)); end
+  def Complex(x, y=T.unsafe(nil), exception: false); end
```
Comment (DarkDimius) [inline rbi/core/kernel.rbi]: could you please duplicate the comment in all of them?

**Comment by DarkDimius** [comment]
Kicked of ci

File: rbi/core/kernel.rbi (hunk 2 above)
Comment (pje) [reply rbi/core/kernel.rbi]: Duplicated!

File: rbi/core/kernel.rbi (hunk 1 above)
Comment (pje) [reply rbi/core/kernel.rbi]: Updated!

**Comment by pje** [comment]
@DarkDimius, I think this is ready to go once the tests pass!

**Comment by DarkDimius** [comment]
Thank you, kicking of build

**Comment by pje** [comment]
@DarkDimius: ready to go!

**Comment by pje** [comment]
I think this is ready, @DarkDimius—is there anything I should do to move this forward?

**Comment by DarkDimius** [comment]
Sorry, Patrick. Didn't get ping and then didn't look into this for quite a while. Landing now

**Pull request merged**